### PR TITLE
Replace in-line code formatter with nightly run

### PR DESCRIPTION
**Pull request by dms1981**
During my investigation into [this issue](https://github.com/ministryofjustice/modernisation-platform/issues/2259) we came to the conclusion that we would replace the existing code-formatter job with one that ran on a nightly basis as a generalised clean-up job.

### CODE REVIEW

File: .github/workflows/format-code.yml [hunk 1]
```diff
@@ -1,19 +1,71 @@
...
+          date=$(date +%Y-%m-%d)
+          branch_name="date-$date"
+          git checkout -b $branch_name
+
+      # MegaLinter
+      - name: Run linter
+        id: ml
+        # You can override MegaLinter flavor used to have faster performances
+        # More info at https://megalinter.io/flavors/
+        uses: oxsecurity/megalinter/flavors/terraform@8fd433c675d27ceca5a61ace53177c1bbfbf7f49
```
Comment (davidkelliott) [inline .github/workflows/format-code.yml:48]: You may want to add the release in a comment after the hash (dependabot will update this as well) to help easily see what version is running

```suggestion
        uses: oxsecurity/megalinter/flavors/terraform@8fd433c675d27ceca5a61ace53177c1bbfbf7f49 #v6.18.0
```

**Review by davidkelliott — APPROVED** [review]
Looks good!

File: .github/workflows/format-code.yml [hunk 2]
```diff
@@ -1,19 +1,78 @@
...
-    permissions:
-      contents: write
+  build:
+    name: MegaLinter
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
-      - uses: ministryofjustice/github-actions/code-formatter@15729e214b8e626847cf4b4596c7635ddb6b1b0a # v10
+      - name: Checkout Code
+        uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c
```
Comment (ewastempel) [inline .github/workflows/format-code.yml:29]: keep the version number pls: # v3.3.0

Comment (dms1981) [reply .github/workflows/format-code.yml]: OK! All done in my most recent commit - [7b900a9](https://github.com/ministryofjustice/modernisation-platform/pull/3105/commits/7b900a94535a641690adfd05539e5df4da96219a)

**Review by ewastempel — APPROVED** [review]
looks good, give it a go?

**Pull request merged**